component/query.libsonnet

```diff
@@ -44,9 +47,69 @@ local query = thanos.query(queryBaseConfig + params.commonConfig + params.query
       type: params.query.serviceType,
     },
   },
+  serviceAccount+: {
+    metadata+: {
+      annotations+: if params.queryRbacProxy.enabled then {
+        'serviceaccounts.openshift.io/oauth-redirecturi.primary': params.queryRbacProxy.redirectUri,
```

Comment:
Again, this feels too OCP-specific to belong in a component which lives in organization projectsyn, any reason why we need openshift-specific annotations for the rbac proxy?

Reply:
This is needed to tell openshift-oauth where to redirect back to after authenticating.

I couldn't make the proxy k8s native... Either it's not possible with kube-rbac-proxy or I was screwing up the config